design-system/react/src/hooks/useStrictedChildren.tsx

```diff
@@ -0,0 +1,31 @@
+import type { ReactNode } from 'react';
+import { Children } from 'react';
+
+export function useStrictedChildren(
```

Comment:
This looks like something that could affect the performance of the application. As if create arrays on every execution, and not caches anything. What do you think?

Reply:
hm, I guess isn't a big thing, but I just added two `useMemo` to prevent `head` and `last` from being on every render. We already have other children `map` operations inside components, isn't something so unusual in fact.